File: apps/android-pos/app/src/main/java/com/pos1/app/ui/screens/KdsScreen.kt
package com.pos1.app.ui.screens

import androidx.compose.animation.animateColorAsState
import androidx.compose.animation.core.LinearEasing
import androidx.compose.animation.core.RepeatMode
import androidx.compose.animation.core.animateFloat
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rememberInfiniteTransition
import androidx.compose.animation.core.tween
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.outlined.AccessTime
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.pos1.app.ui.theme.PosColors
import com.pos1.app.ui.viewmodel.KdsTicketUi
import com.pos1.app.ui.viewmodel.KdsViewModel
import kotlinx.coroutines.delay

/* ═══════════════════════════════════════════════════════════════
   KDS Screen — Dark Premium Kitchen Display  (Ramotion style)
   ═══════════════════════════════════════════════════════════════ */

private val VoidBg      = Color(0xFFF4F6FF)
private val Surface1    = Color(0xFFFFFFFF)
private val Surface2    = Color(0xFFF8F9FF)
private val Ghost       = Color(0x0D000000)
private val GhostBorder = Color(0x12000000)

private data class KdsColDef(val status: String, val labelAr: String, val color: Color)

private val kdsColumns = listOf(
    KdsColDef("NEW",     "جديد",         Color(0xFF3B82F6)),
    KdsColDef("COOKING", "قيد التحضير",  Color(0xFFF59E0B)),
    KdsColDef("READY",   "جاهز",          Color(0xFF10B981)),
    KdsColDef("SERVED",  "تم التقديم",   Color(0xFF64748B)),
)

@Composable
fun KdsScreen(vm: KdsViewModel) {

    LaunchedEffect(Unit) {
        while (true) {
            vm.refreshTickets()
            delay(4000)
        }
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(VoidBg)
            .padding(top = 12.dp),
    ) {
        // ─── Header ───
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp, vertical = 4.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Column {
                Text(
                    "شاشة المطبخ",
                    style = MaterialTheme.typography.headlineMedium,
                    fontWeight = FontWeight.Bold,
                    color = PosColors.Slate900,
                )
                Text(
                    "${vm.tickets.size} تذكرة نشطة",
                    style = MaterialTheme.typography.bodySmall,
                    color = Color(0xFF64748B),
                )
            }
            val slaBreaches = vm.tickets.count { it.elapsedMin > 12 }
            if (slaBreaches > 0) {
                DarkSlaBadge(count = slaBreaches)
            }
        }

        Spacer(Modifier.height(12.dp))

        // ─── Station Filter ───
        LazyRow(
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            contentPadding = PaddingValues(horizontal = 20.dp),
        ) {
            items(vm.stations) { station ->
                val isSelected = station == vm.selectedStation
                Surface(
                    onClick = { vm.selectedStation = station },
                    shape = RoundedCornerShape(20.dp),
                    color = if (isSelected) Color(0xFF7C3AED) else Ghost,
                    border = BorderStroke(1.dp, if (isSelected) Color(0xFF7C3AED) else GhostBorder),
                ) {
                    Text(
                        station,
                        modifier = Modifier.padding(horizontal = 16.dp, vertical = 7.dp),
                        style = MaterialTheme.typography.labelMedium,
                        fontWeight = if (isSelected) FontWeight.Bold else FontWeight.Normal,
                        color = if (isSelected) Color.White else Color(0xFF94A3B8),
                    )
                }
            }
        }

        Spacer(Modifier.height(16.dp))

        // ─── Kanban Board ───
        Row(
            modifier = Modifier
                .fillMaxSize()
                .horizontalScroll(rememberScrollState())
                .padding(horizontal = 12.dp),
            horizontalArrangement = Arrangement.spacedBy(10.dp),
        ) {
            kdsColumns.forEach { col ->
                val colTickets = vm.filteredTickets(col.status)
                DarkKdsColumn(
                    col = col,
                    tickets = colTickets,
                    onStatusChange = { id, next -> vm.updateTicketStatus(id, next) },
                    modifier = Modifier
                        .width(288.dp)
                        .fillMaxHeight(),
                )
            }
        }
    }
}

/* ─── Pulsing SLA Badge ─── */

@Composable
private fun DarkSlaBadge(count: Int) {
    val inf = rememberInfiniteTransition(label = "slaPulse")
    val alpha by inf.animateFloat(
        initialValue = 1f,
        targetValue = 0.35f,
        animationSpec = infiniteRepeatable(tween(700, easing = LinearEasing), RepeatMode.Reverse),
        label = "slaAlpha",
    )
    Box(
        modifier = Modifier
            .alpha(alpha)
            .clip(RoundedCornerShape(10.dp))
            .background(Color(0xFF7F1D1D))
            .border(1.dp, Color(0xFFEF4444).copy(alpha = 0.4f), RoundedCornerShape(10.dp))
            .padding(horizontal = 12.dp, vertical = 6.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(Icons.Outlined.AccessTime, null, modifier = Modifier.size(15.dp), tint = Color(0xFFEF4444))
            Spacer(Modifier.width(6.dp))
            Text(
                "$count تجاوز SLA",
                style = MaterialTheme.typography.labelMedium,
                fontWeight = FontWeight.Bold,
                color = Color(0xFFEF4444),
            )
        }
    }
}

/* ─── KDS Column ─── */

@Composable
private fun DarkKdsColumn(
    col: KdsColDef,
    tickets: List<KdsTicketUi>,
    onStatusChange: (String, String) -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .clip(RoundedCornerShape(16.dp))
            .background(Surface1)
            .border(1.dp, col.color.copy(alpha = 0.18f), RoundedCornerShape(16.dp)),
    ) {
        // ── Column header strip ──
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(col.color.copy(alpha = 0.12f))
                .padding(horizontal = 14.dp, vertical = 11.dp),
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        Modifier
                            .size(9.dp)
                            .clip(CircleShape)
                            .background(col.color)
                            .shadow(4.dp, CircleShape, ambientColor = col.color, spotColor = col.color)
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(
                        col.labelAr,
                        style = MaterialTheme.typography.titleSmall,
                        fontWeight = FontWeight.Bold,
                        color = col.color,
                    )
                }
                Box(
                    Modifier
                        .clip(CircleShape)
                        .background(col.color.copy(alpha = 0.18f))
                        .padding(horizontal = 9.dp, vertical = 3.dp),
                ) {
                    Text(
                        "${tickets.size}",
                        style = MaterialTheme.typography.labelSmall,
                        fontWeight = FontWeight.Bold,
                        color = col.color,
                    )
                }
            }
        }

        // ── Tickets ──
        if (tickets.isEmpty()) {
            Box(
                Modifier
                    .fillMaxWidth()
                    .height(100.dp),
                contentAlignment = Alignment.Center,
            ) {
                Text("لا توجد تذاكر", style = MaterialTheme.typography.bodySmall, color = Color(0xFF475569))
            }
        } else {
            LazyColumn(
                verticalArrangement = Arrangement.spacedBy(8.dp),
                contentPadding = PaddingValues(10.dp),
            ) {
                items(tickets, key = { it.id }) { ticket ->
                    DarkKdsTicketCard(
                        ticket = ticket,
                        currentStatus = col.status,
                        columnColor = col.color,
                        onStatusChange = onStatusChange,
                    )
                }
            }
        }
    }
}

/* ─── KDS Ticket Card ─── */

@Composable
private fun DarkKdsTicketCard(
    ticket: KdsTicketUi,
    currentStatus: String,
    columnColor: Color,
    onStatusChange: (String, String) -> Unit,
) {
    val isSlaBreached = ticket.elapsedMin > 12
    val isWarning     = ticket.elapsedMin > 8

    val borderColor by animateColorAsState(
        when {
            isSlaBreached -> Color(0xFFEF4444)
            isWarning     -> Color(0xFFF59E0B)
            else          -> Color.Transparent
        },
        label = "ticketBorder",
    )

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(Surface2)
            .border(
                width = 1.5.dp,
                color = if (isSlaBreached || isWarning) borderColor else GhostBorder,
                shape = RoundedCornerShape(12.dp),
            )
            .padding(12.dp),
    ) {
        // Header row
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                ticket.orderNo,
                style = MaterialTheme.typography.titleSmall,
                fontWeight = FontWeight.Bold,
                color = PosColors.Slate900,
            )
            Row(verticalAlignment = Alignment.CenterVertically) {
                if (ticket.tableCode != "-") {
                    Box(
                        Modifier
                            .clip(RoundedCornerShape(6.dp))
                            .background(Color(0xFFEFF6FF))
                            .padding(horizontal = 7.dp, vertical = 2.dp),
                    ) {
                        Text(
                            ticket.tableCode,
                            style = MaterialTheme.typography.labelSmall,
                            color = Color(0xFF60A5FA),
                        )
                    }
                    Spacer(Modifier.width(6.dp))
                }
                // Timer badge
                val timerBg = when {
                    isSlaBreached -> Color(0xFF7F1D1D)
                    isWarning     -> Color(0xFF451A03)
                    else          -> Ghost
                }
                val timerColor = when {
                    isSlaBreached -> Color(0xFFEF4444)
                    isWarning     -> Color(0xFFF59E0B)
                    else          -> Color(0xFF94A3B8)
                }
                Box(
                    Modifier
                        .clip(RoundedCornerShape(6.dp))
                        .background(timerBg)
                        .padding(horizontal = 7.dp, vertical = 2.dp),
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Outlined.AccessTime, null, Modifier.size(11.dp), tint = timerColor)
                        Spacer(Modifier.width(3.dp))
                        Text(
                            "${ticket.elapsedMin}د",
                            style = MaterialTheme.typography.labelSmall,
                            fontWeight = FontWeight.Bold,
                            color = timerColor,
                        )
                    }
                }
            }
        }

        if (ticket.items.isNotEmpty()) {
            Spacer(Modifier.height(10.dp))
            // Thin accent divider
            Box(
                Modifier
                    .fillMaxWidth()
                    .height(1.dp)
                    .background(columnColor.copy(alpha = 0.15f))
            )
            Spacer(Modifier.height(8.dp))
            ticket.items.forEach { item ->
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 2.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                ) {
                    Text(
                        "• ${item.qty}× ${item.name}",
                        style = MaterialTheme.typography.bodySmall,
                        color = Color(0xFFCBD5E1),
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis,
                        modifier = Modifier.weight(1f),
                    )
                    if (item.station.isNotBlank()) {
                        Text(
                            item.station,
                            style = MaterialTheme.typography.labelSmall,
                            color = Color(0xFF475569),
                        )
                    }
                }
            }
        }

        // Action button
        val nextStatuses = when (currentStatus) {
            "NEW"     -> listOf("COOKING" to "بدء التحضير")
            "COOKING" -> listOf("READY"   to "جاهز")
            "READY"   -> listOf("SERVED"  to "تم التقديم")
            else      -> emptyList()
        }
        if (nextStatuses.isNotEmpty()) {
            Spacer(Modifier.height(10.dp))
            nextStatuses.forEach { (nextStatus, label) ->
                val btnColor = kdsColumns.firstOrNull { it.status == nextStatus }?.color ?: columnColor
                Button(
                    onClick = { onStatusChange(ticket.id, nextStatus) },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(36.dp),
                    shape = RoundedCornerShape(8.dp),
                    contentPadding = PaddingValues(horizontal = 12.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = btnColor.copy(alpha = 0.85f)),
                ) {
                    Text(label, fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = Color.White)
                    Spacer(Modifier.width(4.dp))
                    Icon(Icons.Default.ChevronRight, null, Modifier.size(15.dp), tint = Color.White)
                }
            }
        }
    }
}

File: ios/ios/UI/Screens/KdsScreen.swift
import SwiftUI

// MARK: ─ KDS Screen (Ramotion Dark Premium Kanban)

struct KdsScreen: View {
    @StateObject private var viewModel: KdsViewModel
    @State private var refreshTimer: Timer?

    init(apiClient: ApiClient) {
        _viewModel = StateObject(wrappedValue: KdsViewModel(apiClient: apiClient))
    }

    fileprivate struct KdsColDef {
        let status: String
        let label: String
        let color: Color
    }

    private let columns: [KdsColDef] = [
        KdsColDef(status: "NEW",     label: "جديد",          color: Color(hex: "3B82F6")),
        KdsColDef(status: "COOKING", label: "قيد التحضير",   color: Color(hex: "F59E0B")),
        KdsColDef(status: "READY",   label: "جاهز",          color: Color(hex: "10B981")),
        KdsColDef(status: "SERVED",  label: "تم التقديم",    color: Color(hex: "64748B")),
    ]

    private var slaBreaches: Int {
        viewModel.tickets.filter { $0.elapsedMin > 12 }.count
    }

    var body: some View {
        ZStack {
            Color(hex: "F4F6FF").ignoresSafeArea()

            VStack(spacing: 0) {
                // ── Header ──────────────────────────────────────────
                HStack(spacing: 12) {
                    VStack(alignment: .trailing, spacing: 4) {
                        Text("شاشة المطبخ")
                            .font(.system(size: 24, weight: .bold))
                            .foregroundColor(Color(hex: "0F172A"))
                        Text("\(viewModel.tickets.count) تذكرة نشطة")
                            .font(.caption)
                            .foregroundColor(Color(hex: "94A3B8"))
                    }

                    Spacer()

                    // SLA breach badge
                    if slaBreaches > 0 {
                        SlaPulseBadge(count: slaBreaches)
                    }

                    // Station filter
                    if viewModel.stations.count > 1 {
                        ScrollView(.horizontal, showsIndicators: false) {
                            HStack(spacing: 6) {
                                ForEach(viewModel.stations, id: \.self) { station in
                                    Button {
                                        viewModel.selectedStation = station
                                    } label: {
                                        let sel = viewModel.selectedStation == station
                                        Text(station)
                                            .font(.system(size: 12, weight: sel ? .bold : .medium))
                                            .foregroundColor(sel ? PosColors.Violet600 : Color(hex: "64748B"))
                                            .padding(.horizontal, 12)
                                            .padding(.vertical, 6)
                                            .background(sel ? PosColors.Violet600.opacity(0.12) : Color.black.opacity(0.04))
                                            .cornerRadius(16)
                                            .overlay(
                                                Capsule().stroke(sel ? PosColors.Violet600.opacity(0.4) : Color.clear, lineWidth: 1)
                                            )
                                    }
                                    .buttonStyle(.plain)
                                }
                            }
                        }
                    }
                }
                .padding(.horizontal, 20)
                .padding(.vertical, 14)

                // ── Kanban board ────────────────────────────────────
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(alignment: .top, spacing: 12) {
                        ForEach(columns, id: \.status) { col in
                            let colTickets = viewModel.filteredTickets(status: col.status)
                            KdsDarkColumn(
                                col: col,
                                tickets: colTickets
                            ) { ticketId, newStatus in
                                Task { await viewModel.updateTicketStatus(ticketId: ticketId, status: newStatus) }
                            }
                            .frame(width: 290)
                        }
                    }
                    .padding(.horizontal, 20)
                    .padding(.bottom, 20)
                }
                .frame(maxHeight: .infinity)
            }
        }
        .onAppear { startAutoRefresh() }
        .onDisappear { stopAutoRefresh() }
        .task { await viewModel.refreshTickets() }
    }

    private func startAutoRefresh() {
        refreshTimer = Timer.scheduledTimer(withTimeInterval: 4.0, repeats: true) { _ in
            Task { await viewModel.refreshTickets() }
        }
    }
    private func stopAutoRefresh() {
        refreshTimer?.invalidate()
        refreshTimer = nil
    }
}

// MARK: ─ SLA Pulse Badge

private struct SlaPulseBadge: View {
    let count: Int
    @State private var pulse = false

    var body: some View {
        HStack(spacing: 6) {
            Circle()
                .fill(Color(hex: "EF4444"))
                .frame(width: 8, height: 8)
                .scaleEffect(pulse ? 1.4 : 1)
                .opacity(pulse ? 0.5 : 1)
                .animation(.easeInOut(duration: 0.7).repeatForever(autoreverses: true), value: pulse)

            Text("\(count) تجاوز SLA")
                .font(.system(size: 12, weight: .bold))
                .foregroundColor(Color(hex: "EF4444"))
        }
        .padding(.horizontal, 12)
        .padding(.vertical, 7)
        .background(Color(hex: "EF4444").opacity(0.1))
        .cornerRadius(20)
        .overlay(Capsule().stroke(Color(hex: "EF4444").opacity(0.25), lineWidth: 1))
        .onAppear { pulse = true }
    }
}

// MARK: ─ KDS Column

private struct KdsDarkColumn: View {
    let col: KdsScreen.KdsColDef
    let tickets: [KdsTicketUi]
    let onStatusChange: (String, String) -> Void

    var body: some View {
        VStack(spacing: 0) {
            // Column header
            HStack {
                HStack(spacing: 7) {
                    Circle().fill(col.color).frame(width: 9, height: 9)
                    Text(col.label)
                        .font(.system(size: 13, weight: .bold))
                        .foregroundColor(col.color)
                }
                Spacer()
                Text("\(tickets.count)")
                    .font(.system(size: 12, weight: .bold))
                    .foregroundColor(col.color)
                    .frame(width: 26, height: 26)
                    .background(col.color.opacity(0.14))
                    .clipShape(Circle())
            }
            .padding(.horizontal, 14)
            .padding(.vertical, 12)
            .background(col.color.opacity(0.07))

            // Tickets
            if tickets.isEmpty {
                VStack {
                    Spacer()
                    VStack(spacing: 8) {
                        Image(systemName: "tray")
                            .font(.system(size: 28))
                            .foregroundColor(Color(hex: "E2E8F0"))
                        Text("لا توجد تذاكر")
                            .font(.caption)
                            .foregroundColor(Color(hex: "94A3B8"))
                    }
                    Spacer()
                }
                .frame(height: 140)
            } else {
                ScrollView(showsIndicators: false) {
                    LazyVStack(spacing: 10) {
                        ForEach(tickets) { ticket in
                            KdsDarkTicketCard(
                                ticket: ticket,
                                currentStatus: col.status,
                                accentColor: col.color,
                                onStatusChange: onStatusChange
                            )
                        }
                    }
                    .padding(10)
                }
            }
        }
        .background(Color(hex: "FFFFFF"))
        .clipShape(RoundedRectangle(cornerRadius: 18))
        .overlay(
            RoundedRectangle(cornerRadius: 18)
                .stroke(col.color.opacity(0.12), lineWidth: 1)
        )
        .frame(maxHeight: .infinity)
    }
}

// MARK: ─ KDS Ticket Card

private struct KdsDarkTicketCard: View {
    let ticket: KdsTicketUi
    let currentStatus: String
    let accentColor: Color
    let onStatusChange: (String, String) -> Void

    private var isSla: Bool { ticket.elapsedMin > 12 }
    private var isWarn: Bool { ticket.elapsedMin > 8 }

    private var timerColor: Color {
        isSla ? Color(hex: "DC2626") : isWarn ? Color(hex: "D97706") : Color(hex: "94A3B8")
    }

    private var nextAction: (status: String, label: String)? {
        switch currentStatus {
        case "NEW":     return ("COOKING", "ابدأ التحضير")
        case "COOKING": return ("READY",   "اجعله جاهزاً")
        case "READY":   return ("SERVED",  "تم التقديم")
        default:        return nil
        }
    }

    var body: some View {
        VStack(alignment: .trailing, spacing: 10) {
            // Top row
            HStack {
                // Table badge
                if ticket.tableCode != "-" {
                    Text(ticket.tableCode)
                        .font(.system(size: 11, weight: .bold))
                        .foregroundColor(Color(hex: "3B82F6"))
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(Color(hex: "3B82F6").opacity(0.12))
                        .cornerRadius(6)
                }

                Spacer()

                Text("#\(ticket.orderNo)")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(Color(hex: "0F172A"))

                // Timer
                HStack(spacing: 4) {
                    Image(systemName: "clock.fill")
                        .font(.system(size: 10))
                    Text("\(ticket.elapsedMin)د")
                        .font(.system(size: 11, weight: .bold))
                        .monospacedDigit()
                }
                .foregroundColor(timerColor)
                .padding(.horizontal, 8)
                .padding(.vertical, 4)
                .background(timerColor.opacity(0.12))
                .cornerRadius(6)
            }

            // Items list
            if !ticket.items.isEmpty {
                VStack(alignment: .trailing, spacing: 5) {
                    ForEach(ticket.items) { item in
                        HStack(spacing: 6) {
                            Spacer()
                            Text(item.name)
                                .font(.system(size: 12))
                                .foregroundColor(Color(hex: "475569"))
                            Text("×\(item.qty)")
                                .font(.system(size: 12, weight: .bold))
                                .foregroundColor(accentColor)
                                .frame(minWidth: 24)
                        }
                    }
                }
                .padding(10)
                .background(Color.black.opacity(0.03))
                .cornerRadius(10)
            }

            // Action button
            if let action = nextAction {
                Button {
                    onStatusChange(ticket.id, action.status)
                } label: {
                    HStack(spacing: 6) {
                        Text(action.label)
                            .font(.system(size: 12, weight: .bold))
                        Image(systemName: "chevron.left")
                            .font(.system(size: 11, weight: .bold))
                    }
                    .foregroundColor(accentColor)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 9)
                    .background(accentColor.opacity(0.12))
                    .cornerRadius(9)
                    .overlay(
                        RoundedRectangle(cornerRadius: 9)
                            .stroke(accentColor.opacity(0.25), lineWidth: 1)
                    )
                }
                .buttonStyle(.plain)
            }
        }
        .padding(12)
        .background(Color(hex: "F8F9FF"))
        .clipShape(RoundedRectangle(cornerRadius: 14))
        .overlay(
            RoundedRectangle(cornerRadius: 14)
                .stroke(
                    isSla ? Color(hex: "EF4444").opacity(0.3) :
                    isWarn ? Color(hex: "F59E0B").opacity(0.2) :
                    Color.black.opacity(0.07),
                    lineWidth: 1
                )
        )
    }
}
